numpy/dtypes.pyi

```diff
@@ -1,43 +1,641 @@
+        "datetime64[m]",
+        "datetime64[s]",
+        "datetime64[ms]",
+        "datetime64[us]",
+        "datetime64[ns]",
+        "datetime64[ps]",
+        "datetime64[fs]",
+        "datetime64[as]",
+    ]: ...
+    @property
+    def str(self) -> L[
+        "<M8", ">M8",
+        "<M8[Y]", ">M8[Y]",
+        "<M8[M]", ">M8[M]",
+        "<M8[W]", ">M8[W]",
+        "<M8[D]", ">M8[D]",
+        "<M8[h]", ">M8[h]",
+        "<M8[m]", ">M8[m]",
+        "<M8[s]", ">M8[s]",
+        "<M8[ms]", ">M8[ms]",
+        "<M8[us]", ">M8[us]",
+        "<M8[ns]", ">M8[ns]",
+        "<M8[ps]", ">M8[ps]",
+        "<M8[fs]", ">M8[fs]",
+        "<M8[as]", ">M8[as]",
+    ]: ...
+
+@final
+class TimeDelta64DType(
+    Generic[_DateTimeUnit_co],
+    _TypeCodes[L["m"], L["m"], L[22]],
+    _NativeOrder,
+    _NBit[L[8], L[8]],
+    _BaseDType[np.timedelta64],
+):
+    # NOTE: `TimeDelta64DType(...)` raises a `TypeError` at the moment
+    # TODO: Once implemented, don't forget to overload on `unit: L["μs"]`.
+    def __new__(cls, unit: _DateTimeUnit_co, /) -> NoReturn: ...
+    @property
+    def flags(self) -> L[0]: ...
+    @property
+    def hasobject(self) -> L[False]: ...
+    @property
+    def isbuiltin(self) -> L[0]: ...
+    @property
+    def name(self) -> L[
+        "timedelta64",
+        "timedelta64[Y]",
+        "timedelta64[M]",
+        "timedelta64[W]",
+        "timedelta64[D]",
+        "timedelta64[h]",
+        "timedelta64[m]",
+        "timedelta64[s]",
+        "timedelta64[ms]",
+        "timedelta64[us]",
+        "timedelta64[ns]",
+        "timedelta64[ps]",
+        "timedelta64[fs]",
+        "timedelta64[as]",
+    ]: ...
+    @property
+    def str(self) -> L[
+        "<m8", ">m8",
+        "<m8[Y]", ">m8[Y]",
+        "<m8[M]", ">m8[M]",
+        "<m8[W]", ">m8[W]",
+        "<m8[D]", ">m8[D]",
+        "<m8[h]", ">m8[h]",
+        "<m8[m]", ">m8[m]",
+        "<m8[s]", ">m8[s]",
+        "<m8[ms]", ">m8[ms]",
+        "<m8[us]", ">m8[us]",
+        "<m8[ns]", ">m8[ns]",
+        "<m8[ps]", ">m8[ps]",
+        "<m8[fs]", ">m8[fs]",
+        "<m8[as]", ">m8[as]",
+    ]: ...
+
+@final
+class StringDType(
+    _TypeCodes[L["T"], L["T"], L[2056]],
+    _NativeOrder,
+    _NBit[L[8], L[16]],
+    # TODO: Replace the (invalid) `str` with the scalar type, once implemented
+    np.dtype[str],  # type: ignore[misc]
+):
+    def __new__(cls, /) -> StringDType: ...
+    def __getitem__(self, key: Any, /) -> NoReturn: ...
+    @property
+    def base(self) -> StringDType: ...
+    @property
+    def fields(self) -> None: ...
+    @property
+    def flags(self) -> L[107]: ...
+    @property
+    def hasobject(self) -> L[True]: ...
+    @property
+    def isalignedstruct(self) -> L[False]: ...
+    @property
+    def isbuiltin(self) -> L[0]: ...
+    @property
+    def isnative(self) -> L[True]: ...
+    @property
+    def metadata(self) -> None: ...
+    @property
+    def name(self) -> L["StringDType128"]: ...
```

Comment:
The name doesn't square with the NBit, I suppose on 32bit system it might be 64 here (same below).